selftests/functional/test_teststmpdir.py

```diff
@@ -0,0 +1,54 @@
+#!/usr/bin/env python
+
+import os
+import sys
+import tempfile
+import shutil
+
+if sys.version_info[:2] == (2, 6):
+    import unittest2 as unittest
+else:
+    import unittest
+
+from avocado.core import exit_codes
+from avocado.utils import process
+from avocado.utils import script
+
+
+basedir = os.path.join(os.path.dirname(os.path.abspath(__file__)), '..', '..')
+basedir = os.path.abspath(basedir)
+
+
+SIMPLE_SCRIPT = """#!/bin/bash
+: ${XXX_TESTS_COMMON_TMPDIR:?}
+"""
+
+
+class TestsTmpDirTests(unittest.TestCase):
+
+    def setUp(self):
+        self.tmpdir = tempfile.mkdtemp(prefix='avocado_' + __name__)
+
+    def run_and_check(self, cmd_line, expected_rc):
+        os.chdir(basedir)
+        result = process.run(cmd_line, ignore_status=True)
+        self.assertEqual(result.exit_status, expected_rc,
+                         "Command %s did not return rc "
+                         "%d:\n%s" % (cmd_line, expected_rc, result))
```

Comment:
I fail to see how this checks it works well. I'd suggest creating a file and listing all files in the test. I'd execute it multiple times and expect the first test to contain one file, second 2 files .... Also I'd combine SIMPLE and INSTRUMENTED tests.

```
avocado --show test run env_simple.sh test.py env_simple.sh
...
TMPDIR=/tmp/avocado-...
file1
env_simple.sh PASS
...
file1 file2
test.py:test_1 PASS
...
file1 file2 file3
test.py:test_2 PASS
...
TMPDIR=/tmp/avocado-...
file1 file2 file3 file4
env_simple.sh
```

and I'd check the `TMPDIR` does not exists after the test and that the files were created in this order (first there should be `file1`, then `file1 file2` ...

Then I'd say it probably works well...